inference-engine/src/mkldnn_plugin/nodes/mkldnn_gather_nd_node.cpp

```diff
@@ -40,180 +37,114 @@ MKLDNNGatherNDNode::MKLDNNGatherNDNode(const std::shared_ptr<ngraph::Node>& op,
     }
-    _errorPrefix = std::string("Layer GatherND with name '") + op->get_friendly_name() + "'";
-
-    if (op->get_input_size() != 2 || op->get_output_size() != 1)
-        IE_THROW() << _errorPrefix << " has invalid number of input/output edges.";
 
-    const auto& dataDims = op->get_input_shape(_dataIndex);
-    const auto& indicesDims = op->get_input_shape(_indicesIndex);
+    if (inputShapes.size() != 2 && outputShapes.size() != 1)
+        THROW_ERROR << "has invalid number of input/output edges.";
 
-    auto gatherNdOp = ngraph::as_type_ptr<const ngraph::op::v5::GatherND>(op);
-    _batchDims = gatherNdOp->get_batch_dims();
-    if (_batchDims >= std::min(dataDims.size(), indicesDims.size()))
-        IE_THROW() << _errorPrefix << " has invalid batch_dims attribute: " << _batchDims;
+    const size_t inputDataRank = getInputShapeAtPort(GATHERND_DATA).getRank();
+    const size_t indicesDimsRank = getInputShapeAtPort(GATHERND_INDEXES).getRank();
 
-    _batchNum = 1lu;
-    for (size_t i = 0; i < _batchDims; i++) {
-        _batchNum *= indicesDims[i];
+    if (auto gatherNdOp = ngraph::as_type_ptr<const ngraph::op::v8::GatherND>(op)) {
+        attrs.batchDims = gatherNdOp->get_batch_dims();
+    } else if (auto gatherNdOp = ngraph::as_type_ptr<const ngraph::op::v5::GatherND>(op)) {
+        attrs.batchDims = gatherNdOp->get_batch_dims();
+    } else {
+        THROW_ERROR << "has support only opset5.";
     }
 
-    _sliceRank = indicesDims[indicesDims.size() - 1];
-    _dataRank = dataDims.size() - _batchDims;
-    if (_sliceRank > _dataRank)
-        IE_THROW() << _errorPrefix << " has invalid inputs shapes.";
-
-    _blockSize = 1;
-    for (size_t i = _sliceRank + _batchDims; i < dataDims.size(); i++) {
-        _blockSize *= dataDims[i];
-    }
-    _batchStep = 1;
-    for (size_t i = _batchDims; i < dataDims.size(); i++) {
-        _batchStep *= dataDims[i];
-    }
+    if (attrs.batchDims >= std::min(inputDataRank, indicesDimsRank))
+        THROW_ERROR << "has invalid batch_dims attribute: " << attrs.batchDims;
 }
 
 void MKLDNNGatherNDNode::initSupportedPrimitiveDescriptors() {
     if (!supportedPrimitiveDescriptors.empty())
         return;
 
-    Precision inDataPrecision = getOriginalInputPrecisionAtPort(_dataIndex);
-    if (!MKLDNNPlugin::one_of(inDataPrecision.size(),
-                              sizeof(PrecisionTrait<Precision::I32>::value_type),
-                              sizeof(PrecisionTrait<Precision::I16>::value_type),
-                              sizeof(PrecisionTrait<Precision::I8>::value_type))) {
-        IE_THROW() << _errorPrefix << " has unsupported 'data' input precision: " << inDataPrecision;
-    }
-
-    Precision indicesPrecision = getOriginalInputPrecisionAtPort(_indicesIndex);
+    Precision inDataPrecision = getOriginalInputPrecisionAtPort(GATHERND_DATA);
+    Precision indicesPrecision = getOriginalInputPrecisionAtPort(GATHERND_INDEXES);
```

Comment:
You dropped inDataPrecision check. In general case it's not a good idea because IE::Precision contains items with non integer number of bytes: INT4, BIN etc. Оbviously your impl will not work with these precisions. Please fix in separate PR